crypt/secret_sharer.go

```diff
@@ -7,8 +7,39 @@ import (
 	"crypto/sha256"
 	"errors"
 	"math/big"
+	"sort"
 )
 
+/////// Share sorting ////////
+type By func(p1, p2 *SecretShare) bool
```

Comment:
I just copied some code I found online... I changed it now to use your code for sorting.